Rebased onto current main and deliberately reconciled the rewritten descendant-PID assertion at exact head `e1450f456e39cea2e153aa6d652c01127569afdb`.

Evidence:
- The lifecycle owner in `extensions/qa-lab/src/posix-command-settlement.ts` settles only after cleanup completes and reports a settlement failure if the original process group remains alive.
- The current-main descendant-PID poll had the same numeric identity-reuse problem as the old PGID poll, so the rebased patch removes only that redundant post-settlement probe while preserving current-main's shared PID-file/wait helpers and all escaped-descendant/parent-signal coverage.
- Production delta: +0/-0. Tests: +2/-4.
- `git diff --check origin/main...HEAD`: pass.
- Autoreview (Codex, branch mode): clean; patch correct 0.99, no actionable findings.
- Focused local Vitest was attempted but this checkout has no `tsx` installation (`Cannot find module 'tsx'`); exact-head CI is the execution proof.

@clawsweeper re-review

--- a/extensions/qa-lab/src/test-file-scenario-command-lifecycle.test.ts
+++ b/extensions/qa-lab/src/test-file-scenario-command-lifecycle.test.ts
@@ -72,7 +72,6 @@ describe.skipIf(process.platform === "win32")("qa scenario command real POSIX li
   it("settles within a bound after the leader writes its final result with inherited stdio open", async () => {
     const root = await mkdtemp(path.join(os.tmpdir(), "qa-command-settlement-"));
     const descendantPidPath = path.join(root, "descendant.pid");
-    let descendantPid: number | undefined;
     spawnMock.mockImplementation((...args: Parameters<NonNullable<typeof actualSpawn.value>>) => {
       if (!actualSpawn.value) {
         throw new Error("real spawn unavailable");
@@ -102,7 +101,7 @@ describe.skipIf(process.platform === "win32")("qa scenario command real POSIX li
         env: process.env,
         timeoutMs: 5_000,
       });
-      descendantPid = await waitForPidFile(descendantPidPath);
+      await waitForPidFile(descendantPidPath);
       const startedAt = Date.now();
       const deadline = new AbortController();
       const result = await Promise.race([
@@ -122,9 +121,6 @@ describe.skipIf(process.platform === "win32")("qa scenario command real POSIX li
         stderr: "",
       });
     } finally {
-      if (descendantPid && isProcessAlive(descendantPid)) {
-        process.kill(descendantPid, "SIGKILL");
-      }
       await rm(root, { force: true, recursive: true });
     }
   });

